Geometry_Engine/Create/PlanarSurface.cs

```diff
@@ -55,19 +55,19 @@ public static PlanarSurface PlanarSurface(ICurve externalBoundary, List<ICurve>
 
             internalBoundaries = internalBoundaries ?? new List<ICurve>();
 
-            foreach (ICurve crv in internalBoundaries)
+            if (internalBoundaries.Any(x => !x.IIsPlanar()))
```

Comment:
I would check the planarity as follows:
1. Check external boundary for planarity - if not, return null
2. Run `Where` Linq query to filter out internal outlines that are coplanar with the external one
3. Do same check as you do in lines 68-71

Lines 58-62 and 73-80 could be removed then.